### PR TITLE
nsolve option

### PR DESCRIPTION
**Pull request by michaeljbrazell**
- added nsolve to mlmgOptions
- added a warning if using agglomeration and nsolve together

### CODE REVIEW

**Review by sayerhs — APPROVED** [review]

**Comment by michaeljbrazell** [comment]
should we update amrex submodule with this PR? or do it later? I can update my submodule locally to test overset masking

File: amr-wind/core/MLMGOptions.H [hunk 1]
```diff
@@ -104,6 +104,14 @@ private:
...
     //! within max iterations
     bool do_fixed_iters{true};
+
+    //! option to use the nsolve bottom solver
+    //! added to help with overset masking
+    bool do_nsolve{false};
+
+    //! set grid size for nsolve
+    //! use a larger number on GPUs
+    int nsolve_grid_size{16};
```
Comment (sayerhs) [inline amr-wind/core/MLMGOptions.H:114]: We should set the default depending on whether GPU is enabled or not?

**Comment by sayerhs** [comment]
We should be fine updating to latest dev snapshot of AMReX

File: amr-wind/core/MLMGOptions.H (hunk 1 above)
Comment (michaeljbrazell) [reply amr-wind/core/MLMGOptions.H:116]: I added an #ifdef

**Review by sayerhs — APPROVED** [review]

**Pull request merged**